src/routes/safe/store/middleware/notificationsMiddleware.js

```diff
@@ -24,14 +26,21 @@ const notificationsMiddleware = (store: Store<GlobalState>) => (next: Function)
         const transactionsList = action.payload
         const userAddress: string = userAccountSelector(state)
         const awaitingTransactions = getAwaitingTransactions(transactionsList, userAddress)
+        const safes = safesMapSelector(state)
+        const safeAddress = safeParamAddressFromStateSelector(state)
+        const currentSafe = safes.get(safeAddress)
 
+        // If the user is not an owner, do not send notifications
+        if (!isUserOwner(currentSafe, userAddress)) {
+          break
+        }
 
-        awaitingTransactions.map((awaitingTransactionsList, safeAddress) => {
+        awaitingTransactions.map((awaitingTransactionsList, txSafeAddress) => {
           const convertedList = awaitingTransactionsList.toJS()
```

Comment:
Not related to the bugfix, but why do we call `.toJS()` here? As I can see this variable is only used to comparing length, which can be used by checking `size` prop on an immutable `List`

Reply:
done